version: '3.7'

networks:
  web:
    external: true
  internal:
    external: false

services:
  traefik:
    container_name: traefik
    image: traefik:v2.0
    ports:
      - "8080:80"
    command:
      - "--api.insecure=true"
      - "--providers.docker=true"
      - "--log.level=ERROR"
      - "--entrypoints.web.address=:80"
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro
    networks:
      - web
      - internal
    labels:
      - traefik.http.routers.api.rule=Host(`traefik.local`)
      - traefik.http.routers.api.service=api@internal
      - traefik.http.routers.api.entrypoints=web

  prometheus:
    image: prom/prometheus
    container_name: prometheus
    command:
      - "--log.level=error"
      - "--config.file=/etc/prometheus/prometheus.yaml"
    expose:
      - 9090
    volumes:
      - ./prometheus/prometheus.yaml:/etc/prometheus/prometheus.yaml
    networks:
      - internal

  grafana:
    image: grafana/grafana:latest
    container_name: grafana
    expose:
      - 3000
    environment:
      - GF_ROOT_URL=http://grafana.local
      - GF_DASHBOARDS_MIN_REFRESH_INTERVAL=1s
      - GF_AUTH_DISABLE_LOGIN_FORM=true
      - GF_AUTH_ANONYMOUS_ENABLED=true
      - GF_AUTH_ANONYMOUS_ORG_ROLE=Admin
    volumes:
      - ./grafana/provisioning/:/etc/grafana/provisioning/
      - ./grafana/plugins/:/var/lib/grafana/plugins/
    networks:
      - internal
    labels:
      - traefik.http.routers.grafana.rule=Host(`grafana.local`)
      - traefik.http.routers.grafana.entrypoints=web